File: medium/0005_longest_palindromic_substring/solution.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        s_len = len(s)

        longest_palindrome = s[0]

        for left in range(len(s) - 1):
            for offset in (1, 2):
                right = left + offset

                if right == s_len or s[left] != s[right]:
                    continue

                while (
                    left > 0
                    and right < s_len - 1
                    and s[left - 1] == s[right + 1]
                ):
                    left -= 1
                    right += 1

                if right - left + 1 > len(longest_palindrome):
                    longest_palindrome = s[left : right + 1]

        return longest_palindrome

File: medium/0005_longest_palindromic_substring/solution.cpp
using namespace std;

#include <string>

class Solution {
public:
  string longestPalindrome(string s) {
    const int sLen = s.size();

    string longestPalindrome{s[0]};
    int left, right;

    for (int i = 0; i < sLen - 1; ++i) {
      for (int offset = 1; offset < 3; ++offset) {
        left = i;
        right = left + offset;

        if (right == sLen || s[left] != s[right])
          continue;

        for (; left > 0 && right < sLen - 1 && s[left - 1] == s[right + 1];
             --left, ++right)
          ;

        if (right - left + 1 > longestPalindrome.size())
          longestPalindrome = s.substr(left, right - left + 1);
      }
    }

    return longestPalindrome;
  }
};
